RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -915,6 +915,105 @@ boolean adjustForBanishIfPossible(monster enemy, location loc)
 	return false;
 }
 
+boolean canFreeRun(monster enemy, location loc)
+{
+	// are there any restrictions on free running?
+	return true;
+}
+
+string findFreeRunSource(boolean inCombat)
+{
+	if(canChangeToFamiliar($familiar[Frumious Bandersnatch]))
+	{
+		// TODO add fam weight buffing
+		int banderRunsLeft = floor((familiar_weight($familiar[Frumious Bandersnatch]) + weight_adjustment()) / 5) - get_property("_banderRunaways").to_int();
+		if(!inCombat)
+		{
+			if(auto_have_skill($skill[The Ode to Booze]) &&
+				banderRunsLeft > 0 &&
+				(have_effect($effect[Ode to Booze]) > 0 || buffMaintain($effect[Ode to Booze])) &&
+				handleFamiliar($familiar[Frumious Bandersnatch]))
+			{
+				// update familiar already called in pre-adv so have to force.
+				use_familiar($familiar[Frumious Bandersnatch]);
+				return "Frumious Bandersnatch";
+			}
+		}
+		else
+		{
+			if(my_familiar() == $familiar[Frumious Bandersnatch] && have_effect($effect[Ode to Booze]) > 0 && banderRunsLeft > 0)
+			{
+				return "Frumious Bandersnatch";
+			}
+		}
+	}
+
+	if(canChangeToFamiliar($familiar[Pair of Stomping Boots]))
+	{
+		// TODO add fam weight buffing
+		// boots and bander share same counter
+		int banderRunsLeft = floor((familiar_weight($familiar[Pair of Stomping Boots]) + weight_adjustment()) / 5) - get_property("_banderRunaways").to_int();
+		if(!inCombat)
+		{
+			if(banderRunsLeft > 0 && handleFamiliar($familiar[Pair of Stomping Boots]))
+			{
+				// update familiar already called in pre-adv so have to force.
+				use_familiar($familiar[Pair of Stomping Boots]);
+				return "Pair of Stomping Boots";
+			}
+		}
+		else
+		{
+			if(my_familiar() == $familiar[Pair of Stomping Boots] && banderRunsLeft > 0)
+			{
+				return "Pair of Stomping Boots";
+			}
+		}
+	}
+
+	if(auto_hasNavelRing())
```

Comment:
may as well also add a check for greatest american pants as well

you could also use peppermint parasol if neither are available, those are only consumed on the 10th attempt

Reply:
i'll add it after this is merged, I'd like to get it in for a LoL run i'm in